assignments/assignment01.rb

```diff
@@ -37,13 +37,11 @@ def number_to_string(n, lang)
 
 # it accepts a string
 # and returns the same string with each word capitalized.
+# assumes titleize does not need to preserve non-word characters in the original string, i.e.,
+# titleize "hEllo,   WORLD" => "Hello World"
 def titleize(s)
-  words = s.split
-  caps = []
-  words.each do |word|
-    caps << word.capitalize
-  end
-  caps.join " "
+  word_array = s.split(/\W+/)
+  (word_array.map { |w| w.capitalize }).join(" ")
```

Comment:
No need for the parens, you can just write:
word_array.map { |w| w.capitalize }.join(" ")